UofS-Pulse-Binfo/analyzedphenotypes
# Phenotypic Data Importer

## Metadata

 - Issue #97



- [x] This PR is dependent upon https://github.com/tripal/tripal/pull/955

Documentation:

## Description
This PR creates a Phenotypic Data Importer which uses the TripalImporter class. This was done to

 - move away from the drag and drop element to the more standle tripal html5
 - allow us to take advantage of file quotas, etc. provided by the Tripal API
 - adhere to Tripal standards

This PR is being made before the loader is complete to meet a loading deadline. See the following:

### Known Issues
1. Does not support photos
2. Does not support crop ontology term mapping
3. When validation errors are encountered in step 3 the theme is altered.

None of these issues impact the loading of core phenotypic data with this importer.

## Testing?




PREP: Create a project and organism. See [here](https://github.com/UofS-Pulse-Binfo/analyzedphenotypes/pull/46) for more including how to insert germplasm for testing and check database results.

1. Go to Admin > Tripal > Data Loaders > Phenotypic Data Importer
2. Upload a file, select an existing project and genus.
3. Clicking next should execute basic validation (file type, column header).
4. Run the tripal job for step 2 which should do more extensive checking including germplasm existing ahead of time.
5. Fill out the trait information on step 3 and submit the job.

Example files in `analyzedphenotypes/tests/example_files`.
